src/http/fastcgi_handler.rs

```diff
@@ -48,104 +46,124 @@ pub(crate) async fn handle_fastcgi(
+    fcgi_params.insert("REQUEST_URI", &request_uri);
+    fcgi_params.insert("SCRIPT_FILENAME", &script_filename);
+    fcgi_params.insert("SCRIPT_NAME", &script_name);
+    fcgi_params.insert("SERVER_ADDR", "127.0.0.1");
+    fcgi_params.insert("SERVER_NAME", "127.0.0.1");
+    fcgi_params.insert("SERVER_PORT", php_port_str.as_ref());
+    fcgi_params.insert("SERVER_PROTOCOL", http_version);
+    fcgi_params.insert("SERVER_SOFTWARE", "Rymfony v0.1.0");
+
+    //
+    // Send all Request HTTP headers to FastCGI,
+    // In the form of "HTTP_..." parameters.
+    //
+    let mut fcgi_headers_normalized = Vec::new();
     for (name, value) in headers.iter() {
         let header_name = format!("HTTP_{}", name.as_str().replace("-", "_").to_uppercase());
 
-        headers_normalized.push((header_name, value.to_str().unwrap()));
+        fcgi_headers_normalized.push((header_name, value.to_str().unwrap()));
     }
-    params.extend(headers_normalized.iter().map(|(k, s)| (k.as_str(), *s)));
+    fcgi_params.extend(fcgi_headers_normalized.iter().map(|(k, s)| (k.as_str(), *s)));
 
-    let output = client
-        .do_request(&params, &mut std::io::Cursor::new(body))
-        .unwrap();
-
-    let stdout: Vec<u8> = output.get_stdout().unwrap();
-    let stdout: &[u8] = stdout.as_slice();
-    let stdout: &str = std::str::from_utf8(stdout).unwrap();
-    let stdout: String = String::from(stdout);
+    let request_body_bytes = hyper::body::to_bytes(request_body).await.unwrap();
 
-    let response_headers_regex = Regex::new(r"(?s)^(.*)\r\n\r\n(.*)$").unwrap();
+    //
+    // Ignition! Do the request!
+    //
+    let fcgi_output = client.do_request(&fcgi_params, &mut std::io::Cursor::new(request_body_bytes)).unwrap();
 
-    let capts: Captures = response_headers_regex.captures(&stdout).unwrap();
+    let fcgi_stderr: Vec<u8> = fcgi_output.get_stderr().unwrap_or(Vec::new());
+    let fcgi_stdout: Vec<u8> = fcgi_output.get_stdout().unwrap_or(Vec::new());
 
-    let headers: &str = &capts[1];
-    let body: String = String::from(&capts[2]);
+    trace!("Received FastCGI response.");
 
-    let single_header_regex = Regex::new(r"^([^:]+):(.*)$").unwrap();
-
-    let headers_normalized: HashMap<HeaderName, HeaderValue> = headers
-        .split("\r\n")
-        .map(|header: &str| {
-            let headers_capts = single_header_regex.captures(header).unwrap();
+    if fcgi_stderr.len() > 0 {
+        error!("FastCGI returned an error:\n{}", std::str::from_utf8(&fcgi_stderr).unwrap());
+    }
 
-            let header_name = &headers_capts[1].as_bytes();
-            let header_value = &headers_capts[2];
+    //
+    // Convert the contents of "fcgi_stdout" into a proper list of HTTP Headers.
+    // Body is supposed to be a bunch of bytes.
+    //
+    let mut normalized_headers = [httparse::EMPTY_HEADER; 80];
+    let mut res = httparse::Response::new(&mut normalized_headers);
+    let headers_len = res.parse(fcgi_stdout.as_slice())?.unwrap();
```

Comment:
It's this specific line that fails. The `res.parse()` one.

For now, I have no idea why.